packages/cardano-services/src/StakePool/TypeormStakePoolProvider/TypeormStakePoolProvider.ts

```diff
@@ -50,9 +74,51 @@ export class TypeormStakePoolProvider extends TypeormProvider implements StakePo
     this.#lastRosEpochs = lastRosEpochs;
   }
 
+  async initializeImpl() {
+    await super.initializeImpl();
+
+    void this.withDataSource((dataSource) => this.getFuse(dataSource));
```

Comment:
```suggestion
    await this.withDataSource((dataSource) => this.getFuse(dataSource));
```

Reply:
This makes the code to hand, the `withDataSource` method must be called once `initialize` procedure completed.

Reply:
Why does it have to be called after initialize is completed?

Reply:
I don't know... I say this empirically: `await`ing for that call makes the code to hang.

Reply:
I would like to understand this before approving the workaround :)

Reply:
It seems because [this](https://github.com/input-output-hk/cardano-js-sdk/blob/51de9c56f451c15bed5b3fdbb136e8b024c53206/packages/cardano-services/src/util/TypeormService/TypeormService.ts#L58) is called in `startImpl` which runs later than `initializeImpl`.

Calling `getFuse` during the second event rather than during the first one, should fix.
4d200a9928

Reply:
So the query doesn't work until some other service is `initialize`d?

Reply:
Oh ok I understand it now. I agree moving it to `start` is a good solution :handshake:

Reply:
I would say until the service itself is not `start`ed; `withDataSource` is implemented in `TypeormService` which is in the hierarchy chain of `TypeormStakePoolProvider`.